Modern UI - Always show the seeking ControlBar during ad playback

123mpozzi opened the pull request:
## Description
<!-- Add a short description about the changes -->

<img width="718" height="419" alt="image" src="https://github.com/user-attachments/assets/7cd9e9c5-44a3-4241-beda-e76b12c27190" />

> ControlBar bottom is hidden


With the ad screen UI re-design, we will always show the bottom part of the **control bar** (seek bar, ad duration info), together with the ad skip button, during the ad playback.

The ad skip button is already always visible, so this PR only handles the control bar.

### Changes
- Always show the ControlBar if an ad is playing
- Resume seek bar updates when ad starts
- Add a new hide animation mixin: it will make the element fade by sliding down
  - Use it for the hideable part of the ControlBar, its bottom part


## Checklist (for PR submitter and reviewers)
<!-- A PR with CHANGELOG entry will be released automatically -->
<!-- This is required and should be added in every case -->
- [x] `CHANGELOG` entry - `not needed since base is modern ui`

123mpozzi committed:
Always show the seeking ControlBar during ad playback

- By preventing onControlsHide in ControlBar.ts
- and resuming seek bar updates in SeekBar.ts

stonko1994 reviewed (commented):
The actual design idea was to only show the elements on the leve of the seekbar during ad playback. So to hide the `control-bar-bottom` elements but keep the `control-bar-top` elements.

123mpozzi committed:
Hide the bottom part of the control bar only, keep seek bar and time

123mpozzi committed:
Extract styles specific to ad control bat bottom into its own file

123mpozzi committed:
Add ad control bar bottom

123mpozzi committed:
Revert comment

123mpozzi committed:
Revert boolean expression

123mpozzi committed:
Merge styles and reduce nesting from 3 to 2 levels

123mpozzi committed:
Improve comments

123mpozzi committed:
Merge branch 'feature/modern-ui-base' into feature/modern-ui-ad-interface-make-seekbar-and-button-always-visible

123mpozzi commented:
> The actual design idea was to only show the elements on the leve of the seekbar during ad playback. So to hide the `control-bar-bottom` elements but keep the `control-bar-top` elements.

I see, I addressed this in the recent commits. Now only the bottom part gets hidden

stonko1994 reviewed (commented):
There seems to be some issues with the hiding 🤔 When testing it, it sometimes doesn't hide at all and when it hides it is not animated so it's pretty distracting.

123mpozzi committed:
Hide element sliding down

123mpozzi committed:
Extract slide down hide animation into mixin

123mpozzi committed:
Make slide down class to use our hide/show convention

123mpozzi committed:
Prevent nesting 3 times, and add ending new line

123mpozzi committed:
Use 2 as max scss nesting level to fix linter warning

123mpozzi committed:
Add missing end of line

123mpozzi commented:
> There seems to be some issues with the hiding 🤔 When testing it, it sometimes doesn't hide at all and when it hides it is not animated so it's pretty distracting.

I've addressed the hiding animation in recent commits:
- 18e68f23055809acf213009c2c42cc400e4a7012
- c47645ad98e4f13a769cfac3fd6a6503e39eefd7

It is a generic style and re-usable via mixins 🙂

123mpozzi committed:
Revert changes due to linter since there will always be a warning here

123mpozzi commented:
> There seems to be some issues with the hiding 🤔 When testing it, it sometimes doesn't hide at all and when it hides it is not animated so it's pretty distracting.

Since this is a bug unrelated to the changes, but instead already present on `feature/modern-ui-base`, I'll handle it in a different PR.

Bug is: `controlsHide` event not being emitted in ad screen without a mouse enter/leave event being triggered at least once

123mpozzi committed:
Use hideable control bar bottom for small screen devices

123mpozzi committed:
Merge branch 'feature/modern-ui-base' into feature/modern-ui-ad-interface-make-seekbar-and-button-always-visible

stonko1994 reviewed (commented):
<img width="1552" height="1237" alt="Screenshot 2025-07-31 at 15 16 39" src="https://github.com/user-attachments/assets/fa3ccd7b-5bf8-432a-84f9-a6f9da9fc0f7" />

There seems to be an issue that the seekbar isn't moving down once the control bar is gone. 🤔 

I think we need to implement a Javasript based animation approach here and eventually set `display: none` in css to achieve this.

123mpozzi committed:
Slide down the top part of control bar once the bottom part gets hidden

123mpozzi committed:
Extract falling/slide down (non hidden) animation style into mixin

123mpozzi commented:
> <img alt="Screenshot 2025-07-31 at 15 16 39" width="1552" height="1237" src="https://private-user-images.githubusercontent.com/6216959/473007760-fa3ccd7b-5bf8-432a-84f9-a6f9da9fc0f7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM5NzI5MTcsIm5iZiI6MTc1Mzk3MjYxNywicGF0aCI6Ii82MjE2OTU5LzQ3MzAwNzc2MC1mYTNjY2Q3Yi01YmY4LTQzMmEtODRmOS1hNmY5ZGE5ZmMwZjcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDczMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA3MzFUMTQzNjU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjllYWM5NDNhOWUwZWJiZGY1MWIwYjExYjJlMWZjYTQwZDViODczMzMwN2YxZGZmNzhkNmM2OGM1MGYzZWU4NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.fKMJ5lVP0wWoCabXDublvKRP_jiAZg1SDS3zx6Zdz-I">
> There seems to be an issue that the seekbar isn't moving down once the control bar is gone. 🤔
> 
> I think we need to implement a Javasript based animation approach here and eventually set `display: none` in css to achieve this.

Ah true.
Address it in the recent commits:
- dad2b151406a8469d193e08f4f12db7ce314dd14
- 0b13ec3b7cb2880f08d4e5860db592ceec516ce0

123mpozzi committed:
Slide down skip button as well

123mpozzi committed:
Use better semantic naming for end of transition slide-down mixin

123mpozzi committed:
Update local var to match class name change

123mpozzi committed:
Ensure ControlBar stays hidden when transitioning back to main content

123mpozzi committed:
Fix controls are shown when transitioning back to main content from ad

The issue is that our UI has multiple layouts existing in the DOM at
the same time (e.g. Generic layout and Ad layout).
When the Ad layout goes on, the Generic one is just hidden, but all
its element are there, in the DOM.

So we cannot prevent all hiding of the ControlBar, since we would also
prevent the hiding of the Generic layout's ControlBar.
And we do not want that; in fact, the main content ControlBat should
be hidden when we resume main content from the ad.

I fixed this by differentiating which bar should actually prevent
the hiding.

Note: hiding on AdBreakFinished is useless, as this is a bigger
concern. That's why I removed it.

stonko1994 commented on src/scss/_mixins.scss:
Please check the linter issues in this file

123mpozzi replied:
Address in b09b923df549201017de3506411dbb0bb1ee1b7b

stonko1994 commented on src/scss/components/ads/_ad-control-bar-bottom.scss:
Please check the linter issues in this file

123mpozzi replied:
This warning isn’t easily solvable, I actually looked into it in earlier commits (e.g. 3fd9c637eb73142acbec64c38d7b3744547855a2). Issue is it makes other warnings pop up.

Given the effort involved and that it’s just a linter *warning*, I personally don’t think it’s worth addressing right now

stonko1994 commented on src/ts/components/ads/AdControlBarBottom.ts:
```diff
@@ -0,0 +1,73 @@
+      this.hide();
+    });
+  }
+}
```
Please add an empty line at EOF.

123mpozzi replied:
Address in 7617fc20a11f21f881e5f429599097cd352a385c

stonko1994 commented on src/ts/components/seekbar/SeekBar.ts:
I think those changes here in this file are not needed anymore or? 🤔

123mpozzi replied:
They are necessary, otherwise the seekbar will not show progress

123mpozzi committed:
Remove trailing whitespaces and order elements

123mpozzi committed:
Add end of line at end of file

stonko1994 commented on src/scss/components/ads/_ad-control-bar-bottom.scss:
```diff
@@ -0,0 +1,20 @@
+      display: flex;
+      margin: .5em 0;
+
+      .#{$prefix}-ui-volumeslider {
```
Please fix or silence the linter warning with a meaningfull message.

123mpozzi replied:
7570139eccfca95540318359c837dffea781d5a6

123mpozzi committed:
Ignore nesting deep rule

123mpozzi committed:
Use correct linter and add meaningful message
